### PR TITLE
Perform Paligo updates - CLM-28670

### PR DESCRIPTION
**Pull request by eduard-tita**
Jira: https://sonatype.atlassian.net/browse/CLM-28670
Build: https://jenkins.ci.sonatype.dev/job/integrations/job/jenkins/job/feature-snapshots/job/CLM-28670-paligo-update/

### CODE REVIEW

File: Jenkinsfile.sonatype.deploy [hunk 1]
```diff
@@ -13,6 +13,9 @@ def jira = [
 // Get jenkins plugin version
 def version = getVersionFromBuildName(env.releaseBuild_NAME)
 
+def segments = parseVersionString(getVersionFromBuildName(env.releaseBuild_NAME))
```
Comment (koraytugay) [inline Jenkinsfile.sonatype.deploy:16]: Can be

```suggestion
def segments = parseVersionString(version)
```

`version` coming from line 14?

File: Jenkinsfile.sonatype.deploy [hunk 2]
```diff
@@ -13,6 +13,9 @@ def jira = [
...
+def confluenceVersion = segments.join('.')
```
Comment (koraytugay) [inline Jenkinsfile.sonatype.deploy:17]: Isn't this going to be same with the `version` variable above? I do not understand why we need this.

Comment (eduard-tita) [reply Jenkinsfile.sonatype.deploy:17]: For a version string like `1.2.3-01`, `parseVersionString` will return only the major, minor and patch segments. So the `confluenceVersion` will be `1.2.3`.

Comment (koraytugay) [reply Jenkinsfile.sonatype.deploy:17]: I see, thank you.

File: Jenkinsfile.sonatype.deploy [hunk 3]
```diff
@@ -82,6 +85,27 @@ pipeline {
...
+            [
+                pattern:     /<para[^>]*>[^<]+<\/para>/,
+                replacement: /<para>${confluenceVersion}<\/para>/
+            ]]
+        // paligo: update plugin's links in the download and compatibility page
+        // https://sonatype.atlassian.net/wiki/spaces/HELP/pages/270172330/Automation+with+Paligo#Jenkins-Plugin-Download
+        updatePaligoContent contentId: '170410', updates: [
+            [
+                pattern:     /nexus-jenkins-plugin-[-\d\.]+(.hpi)?/,
+                replacement: /nexus-jenkins-plugin-${version}.hpi/
```
Comment (koraytugay) [inline Jenkinsfile.sonatype.deploy:105]: The data we have in the example is:

```
<link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="https://download.sonatype.com/integrations/jenkins/nexus-jenkins-plugin-3.19.0-01.hpi">nexus-jenkins-plugin-3.19.1</link>
```

Note how it does not have the `.hpi` in the body of the XML tag:
```
<link>nexus-jenkins-plugin-3.19.1</link>
```

I think this replacement will add the `.hpi` there. Is it what we wanted?

**Review by koraytugay — APPROVED** [review]
+1 with a comment/question

File: Jenkinsfile.sonatype.deploy (hunk 3 above)
Comment (eduard-tita) [reply Jenkinsfile.sonatype.deploy:105]: Yes. In other places where we offer the artifact for download we include the file extension. (e.g. Bamboo plugin). I think we should do that everywhere.

**Review by juanmafabbri — APPROVED** [review]
+1

**Pull request merged**